lib/src/providers/VernaProvider/schemaFormActions.tsx

```diff
@@ -0,0 +1,125 @@
+import { ObjectFieldTemplateProps } from '@rjsf/core';
+import React, { Reducer, useReducer } from 'react';
+import { VernaSchemaType } from ':/types/rjsf';
+import { TranslationType } from ':/types/translations';
+import { Maybe } from ':/types/utils';
+import { ReducerAction, VernaActionsEnum } from ':/types/VernaProvider';
+import ShowCaseWidgetProps from ':/types/Widgets';
+import { reduceSchema } from ':/utils/schema';
+
+export function schemaReducer(state: VernaSchemaType, action: ReducerAction): VernaSchemaType {
+  if (action.error) {
+    console.error(
+      'An error happened while applying a modification on the schema.\n',
+      action.payload,
+    );
+    return state;
+  } else {
+    const selector: Maybe<string> = action.meta?.selector;
+
+    if (selector) {
+      const reducedSchema: Maybe<VernaSchemaType> = reduceSchema(
+        {
+          schema: state.schema?.properties?.[selector],
+          translationSchema: state.translationSchema,
+          uiSchema: state.uiSchema?.[selector],
+        },
+        action,
+      );
+      if (!reducedSchema) throw new Error('Error, something went wrong while modifying the schema');
+
+      return {
+        schema: { ...state.schema, [selector]: reducedSchema.schema },
+        translationSchema: reducedSchema.translationSchema,
+        uiSchema: { ...state.uiSchema, [selector]: reducedSchema.uiSchema },
+      };
+    } else return reduceSchema(state, action) || state;
+  }
+}
+
+interface FormSchemaProps {
+  Section: React.FunctionComponent<ObjectFieldTemplateProps>;
+  defaultSchema: VernaSchemaType;
+  selector: Maybe<string>;
+}
+
+export default function useSchemaReducer({ Section, defaultSchema, selector }: FormSchemaProps) {
+  const [schema, _schemaDispatch] = useReducer<Reducer<VernaSchemaType, ReducerAction>>(
+    schemaReducer,
+    defaultSchema,
+  );
+
+  /**
+   * Overload of the schema dispatch to add the selector automatically
+   * @param action given to the dispatch
+   */
+  function schemaDispatch(action: ReducerAction) {
+    _schemaDispatch({ ...action, meta: { ...action.meta, selector } });
+  }
+
+  function addVernaSection(idPreviousSection: string) {
```

Comment:
You could wrap these methods within a `useCallback` to prevent to create a new function at each render.